pytorch_lightning/trainer/distrib_parts.py

```diff
@@ -640,4 +640,8 @@ def determine_root_gpu_device(gpus):
     # set root gpu
     root_gpu = gpus[0]
 
+    # set cuda device to root gpu
+    root_device = (torch.device("cuda", root_gpu) if root_gpu >= 0 else torch.device("cpu"))
+    torch.cuda.set_device(root_device)
+
     return root_gpu
```

Comment:
To me it looks like this should not go here, because the method is called "determine_root_gpu_device", and the added code also checks if the device is cpu. This code should probably go to the place where we call `determine_root_gpu_device`.

Reply:
Where do you suggest to place this?